pkg/eks/api.go

```diff
@@ -48,18 +46,26 @@ import (
 // ClusterProvider stores information about the cluster
 type ClusterProvider struct {
 	// core fields used for config and AWS APIs
-	Provider api.ClusterProvider
+	AWSProvider api.ClusterProvider
+	// KubernetesProvider offers helper methods to handle Kubernetes operations
+	KubernetesProvider KubeProvider
```

Comment:
If we embed this field, we wouldn't have to reference this field via `ctl` everywhere and it would also make the diff smaller. 
```suggestion
	KubeProvider
```

Reply:
Hmmm, not a bad idea. I guess it would result in a smaller diff since I don't have to call it explicitly everywhere. 🤔

Reply:
Hm, not that much :) The AWSProvider rename touched more. :) But I've done it anyways.